### PR TITLE
chore: use click command groups for bento subcommand

### PR DESCRIPTION
**Pull request by sauyon**
This seems to be a better (and type-checker friendly) way to add subcommand groups.

### CODE REVIEW

File: src/bentoml_cli/cli.py [hunk 1]
```diff
@@ -8,52 +8,49 @@
...
-    return bentoml_cli
-
-
-cli = create_bentoml_cli()
+    sys.stdout.reconfigure(encoding="utf-8")  # type: ignore
 
 
 if __name__ == "__main__":
-    cli()
+    bentoml_cli()
```
Comment (aarnphm) [inline src/bentoml_cli/cli.py:56]: we will probably need to do two things here:
1. update the cli_events_map for our tracking
2. update the project_entrypoint in project.toml

Comment (sauyon) [reply src/bentoml_cli/cli.py:56]: Oh, this is actually a refactor error, let me revert that...

File: src/bentoml_cli/bentos.py [hunk 2]
```diff
@@ -47,298 +59,288 @@ def parse_delete_targets_argument_callback(
...
-                ),
-            }
-            for bento in sorted(
-                bentos, key=lambda x: x.info.creation_time, reverse=True
+bento_store = BentoMLContainer.bento_store.get()
+yatai_client = BentoMLContainer.yatai_client.get()
+
+@click.group('bentos')
+def bento_commands():
+    pass
```
Comment (aarnphm) [inline src/bentoml_cli/bentos.py:67]: We can add a quick docstring for this to show it on the CLI

Comment (sauyon) [reply src/bentoml_cli/bentos.py:67]: 👍 should do this.

File: src/bentoml_cli/bentos.py [hunk 3]
```diff
@@ -51,318 +63,307 @@ def parse_delete_targets_argument_callback(
...
-            if _usage_event_debugging():
-                # NOTE: Since we are logging all of the trackintg id to stdout
-                # We will prefix the tag with __tag__ and we can use regex to correctly
-                # get the tag from 'bentoml.bentos.build|build_bentofile'
-                click.echo(f"__tag__:{bento.tag}")
-            else:
-                # Current behaviour
-                click.echo(bento.tag)
-            return bento
+@click.group("bentos")
```
Comment (aarnphm) [inline src/bentoml_cli/bentos.py:70]:
```suggestion
@click.group("bentos", cls=BentoMLCommandGroup)
```

**Comment by sauyon** [comment]
Closing, probably not worth the merge.

**Pull request closed without merging**